src/pybind/mgr/diskprediction/agent/metrics/__init__.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import absolute_import
+
+import time
+
+from .. import BaseAgent
+
+
+class MetricsAgent(BaseAgent):
+
+    def check_database(self):
+        try:
+            resp = self._command.show_database()
+            data = resp.json()
+            if resp.status_code >= 200 and resp.status_code < 300:
+                rc = data.get('results', [])
+                if rc:
+                    series = rc[0].get('series', [])
+                    if series:
+                        values = series[0].get('values', [])
+                        if self._command.dbname not in values:
+                            self._command.create_database()
+                        return True
+            return False
+        except Exception as e:
+            self._logger.error(str(e))
+            return False
+
+    def show_summary(self, status_info):
+        try:
+            if status_info:
+                measurement = status_info['measurement']
+                success_count = status_info['success_count']
+                failure_count = status_info['failure_count']
+                total_count = success_count + failure_count
+                display_string = \
+                    "%s agent stats in total count: %s, success count: %s, failure count: %s."
+                self._logger.info(
+                    display_string % (measurement, total_count, success_count, failure_count)
+                )
+        except Exception as e:
+            self._logger.error(str(e))
+
+    def query_info(self, sql):
+        return self._command.query_info(sql)
+
+    def _run(self):
+        collect_data = self.data
+        result = {}
+        if collect_data:
+            status_info = dict()
+            status_info['measurement'] = None
+            status_info['success_count'] = 0
+            status_info['failure_count'] = 0
+            for dp_data in collect_data:
+                measurement = dp_data.__class__.measurement
+                if not status_info['measurement']:
+                    status_info['measurement'] = measurement
+                tag_list = []
+                field_list = []
+                for name in dp_data.tags:
+                    tag = '%s=%s' % (name, dp_data.tags[name])

```

Comment:
Consider to use ``'%s=%s'.format(name, dp_data.tags[name])`` instead as done in the Dashboard python code. This applies to the whole code of this PR.